diff --git a/vendor/knative.dev/pkg/webhook/resourcesemantics/defaulting/controller.go b/vendor/knative.dev/pkg/webhook/resourcesemantics/defaulting/controller.go
index ba50005d7..31cfda0ea 100644
--- a/vendor/knative.dev/pkg/webhook/resourcesemantics/defaulting/controller.go
+++ b/vendor/knative.dev/pkg/webhook/resourcesemantics/defaulting/controller.go
@@ -113,7 +113,7 @@ func newController(ctx context.Context, name string, optsFunc ...OptionFunc) *co
 	logger := logging.FromContext(ctx)
 	controllerOptions := wopts.ControllerOptions
 	if controllerOptions == nil {
-		const queueName = "DefaultingWebhook"
-		controllerOptions = &controller.ControllerOptions{WorkQueueName: queueName, Logger: logger.Named(queueName)}
+		// TODO: https://github.com/knative/pkg/issues/2418
+		controllerOptions = &controller.ControllerOptions{WorkQueueName: name, Logger: logger.Named(name)}
 	}
 	c := controller.NewContext(ctx, wh, *controllerOptions)
